### PR TITLE
Fix a bug caused by a change in auth_handler function

### PR DESCRIPTION
**Pull request by negzi**
Fix a bug caused by a change in auth_handler function
also fix the relevant unit test cases

### CODE REVIEW

**Comment by erikjohnston** [comment]
@matrixbot ok to test

File: synapse/handlers/register.py [hunk 1]
```diff
@@ -388,8 +388,8 @@ def get_or_create_user(self, localpart, displayname, duration_seconds):
 
         user = UserID(localpart, self.hs.hostname)
         user_id = user.to_string()
-        auth_handler = self.hs.get_handlers().auth_handler
-        token = auth_handler.generate_short_term_login_token(user_id, duration_seconds)
+        token = self.auth_handler().generate_short_term_login_token(
+            user_id, duration_seconds)
```
Comment (erikjohnston) [inline synapse/handlers/register.py]: Code style wise we prefer the closing parenthesis to be on a new line:

```python
token = self.auth_handler().generate_short_term_login_token(
    user_id, duration_seconds
)
```

**Comment by erikjohnston** [comment]
@negzi If you want you can also add an integration test to [sytest](https://github.com/matrix-org/sytest/) (though it is in perl). It should be relatively easy, and would probably look something like: https://github.com/matrix-org/sytest/blob/develop/tests/10apidoc/01register.pl#L37-L62

**Comment by erikjohnston** [comment]
Also, this LGTM, just waiting for tests to finish before merging.

**Comment by erikjohnston** [comment]
Oh! And can you sign off the PR too please :)

**Comment by negzi** [comment]
Signed-off-by: Negar Fazeli negar.fazeli@ericsson.com

File: synapse/handlers/register.py (hunk 1 above)
Comment (negzi) [inline synapse/handlers/register.py]: oh i haven't seen this comment will fix it after adding the integration test if that is OK :)

Comment (erikjohnston) [inline synapse/handlers/register.py]: It's fine, I'll probably clear it up myself since it such a small thing. It was more just an FYI for next time :)

**Pull request merged**